<a href="https://colab.research.google.com/github/mrsferret/Code-Division/blob/main/Worksheets/10_2_Movies_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Movies Mini-project
---

In the previous worksheet you converted an SQL relational database to a single pandas dataframe and downloaded it. You will be analysing it today.

If you were unable to download the file, there is a copy located here: 

"https://github.com/lilaceri/Working-with-data-/blob/main/Data%20Sets%20for%20code%20divisio/movies.csv?raw=true"



### Inspect the dataset 
---

In [35]:
import pandas as pd

def inspect_data(df):

  display(df.describe())
  display(df.head())
  display(df.info())

  # check the data for NA/NaN/missing values across the dataframe
  display("Null Values: ", df.isnull().values.any())

  #============================================================================
  # null values do exist in the dataset and df.info() indicates that column "rank"
  # is only column with them
  #============================================================================
  display("Null Values: ", df.isnull().values.any())
  '''' is any missing values in dataframe '''
  display(df.isnull())

  #============================================================================
  # lets look more closely at the rank column as we know it contains null values
  # check what unique values are stored in this column and its count
  #============================================================================
  print("\nUnique values from column rank...\n",df['rank'].unique())
  print("\nCount unique values from a column ...\n",df['rank'].nunique())

  ### get count of missing values of a particular column
 
  #display ("Count of missing values in rank col: ", df.rank.isnull().sum())
  #display (df.rank.isnull().sum())

  display("count of missing values: ", df.isnull().sum())
  display (df.isnull().sum()/len(df)*100)

  #========================================================================
  # should only be 1 row per movie so check for duplicates
  #========================================================================
  new_val = df[df.duplicated(['name','last_name','year'])]
  print("Duplicate Values")
  print(new_val)
  
  

# program starts here
url = "https://github.com/lilaceri/Working-with-data-/blob/main/Data%20Sets%20for%20code%20divisio/movies.csv?raw=true"

movie_df = pd.read_csv(url)

inspect_data(movie_df)

Unnamed: 0           year           rank
count  368894.000000  368894.000000  113376.000000
mean   184446.500000    1969.442222       5.995124
std    106490.669438      29.082051       1.540330
min         0.000000    1888.000000       1.000000
25%     92223.250000    1947.000000       5.100000
50%    184446.500000    1976.000000       6.200000
75%    276669.750000    1996.000000       7.100000
max    368893.000000    2007.000000       9.900000

Unnamed: 0 first_name    last_name                      name  year  rank  \
0           0        Les  12 Poissons      Rondo pour trompette  2002   NaN   
1           1    Lejaren     a'Hiller  Sleep of Cyma Roget, The  1920   NaN   
2           2   Khairiya    A-Mansour          Churches in Iraq  1988   NaN   
3           3    Ricardo     A. Solla                  Foto, La  2001   NaN   
4           4    Ricardo     A. Solla                  Foto, La  2001   NaN   

         genre  
0        Short  
1        Drama  
2  Documentary  
3        Drama  
4        Short

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 368894 entries, 0 to 368893
Data columns (total 7 columns):
 #   Column      Non-Null Count   Dtype  
---  ------      --------------   -----  
 0   Unnamed: 0  368894 non-null  int64  
 1   first_name  368894 non-null  object 
 2   last_name   368894 non-null  object 
 3   name        368894 non-null  object 
 4   year        368894 non-null  int64  
 5   rank        113376 non-null  float64
 6   genre       368894 non-null  object 
dtypes: float64(1), int64(2), object(4)
memory usage: 19.7+ MB


None

'Null Values: '

True

'Null Values: '

True

Unnamed: 0  first_name  last_name   name   year   rank  genre
0            False       False      False  False  False   True  False
1            False       False      False  False  False   True  False
2            False       False      False  False  False   True  False
3            False       False      False  False  False   True  False
4            False       False      False  False  False   True  False
...            ...         ...        ...    ...    ...    ...    ...
368889       False       False      False  False  False   True  False
368890       False       False      False  False  False   True  False
368891       False       False      False  False  False  False  False
368892       False       False      False  False  False   True  False
368893       False       False      False  False  False   True  False

[368894 rows x 7 columns]


Unique values from column rank...
 [nan 8.3 6.8 6.3 6.  4.3 6.9 6.7 4.4 3.3 4.8 4.5 5.  4.2 4.1 3.5 7.5 5.6
 7.8 7.3 2.7 3.9 7.1 7.4 5.9 8.5 5.7 3.8 6.4 8.1 7.  8.6 7.2 6.1 4.6 6.5
 5.2 5.8 3.2 2.4 2.5 5.1 6.6 4.7 7.7 6.2 5.5 2.9 9.2 8.2 8.9 5.4 2.2 7.6
 5.3 4.9 3.4 2.  1.9 2.6 1.7 1.5 3.1 2.1 1.  1.6 3.6 1.8 9.  4.  3.  8.
 9.3 9.1 1.4 7.9 2.8 8.7 9.4 9.8 9.7 8.4 8.8 3.7 1.1 2.3 9.9 1.3 9.6 9.5
 1.2]

Count unique values from a column ...
 90


'count of missing values: '

Unnamed: 0         0
first_name         0
last_name          0
name               0
year               0
rank          255518
genre              0
dtype: int64

Unnamed: 0     0.000000
first_name     0.000000
last_name      0.000000
name           0.000000
year           0.000000
rank          69.265968
genre          0.000000
dtype: float64

Duplicate Values
        Unnamed: 0          first_name last_name  \
4                4             Ricardo  A. Solla   
7                7  Kodanda Rami Reddy        A.   
9                9  Kodanda Rami Reddy        A.   
10              10  Kodanda Rami Reddy        A.   
11              11  Kodanda Rami Reddy        A.   
...            ...                 ...       ...   
368879      368879              Memduh        Ün   
368881      368881              Memduh        Ün   
368883      368883              Memduh        Ün   
368886      368886              Memduh        Ün   
368893      368893              Yüksel     Ünsal   

                                 name  year  rank    genre  
4                            Foto, La  2001   NaN    Short  
7                          Anubandham  1984   NaN  Romance  
9       Athaku Yamudu Ammayiki Mogudu  1989   NaN   Comedy  
10      Athaku Yamudu Ammayiki Mogudu  1989   NaN    Drama  
11      Athaku Yamudu Ammayiki Mogudu  1989   NaN   F



---


**Inspecting the Data Notes**

---


1.   On initial look at the data, I can see that the dataset has 368894 recs. From the .info(), the only column that contains null data is the "rank" column. This will need to be cleaned. The first column in the dataframe is called "**Unnamed: 0**" and on initial inspection it looks like it is just a record number. Will need to look at this further.

2.   **Rank Column** - Looking more closely at the rank column which contains null values, we can see that there are 255518 records with a missing rank value and this represents nearly 70% of the total number of recs. There are 90 unique values in this column (including nan) with a max value of 9.9 and a minimum of 1. This leaves us with the question of what to do with the rank column? Options are to a) drop the column entirely from the dataframe b) somehow impute a sensible value for the empty fields c) remove only those records that have an empty rank value.

1.  **Duplicate row check** - when checking for duplicates, no Movie/Director/Year combination should have more than 1 row. 
2.   List item


2.   List item



 






### Clean the dataset 
---


### How many movies of each genre are there?
---

### Which director has the highest ranked movies?
---

### How many movies have ranks of over 9?
---


### Plot a bar chart of mean rank and genre
---

### Plot a pie chart of how many movies of each genre there are 
---


### Plot a graph showing the mean Rank for each year 

### What else can you find out from this dataset?
---
Make a plan of 3 further things you can do to interrogate and analyse this dataset

Type your answer here  

1.  
2.  
3.   

### Complete the tasks you have set out in the exercise above. 
---


# Reflection
----

## What skills have you demonstrated in completing this notebook?

Your answer:

## What caused you the most difficulty?

Your answer: